Sources/Apodini/Request/DynamicProperty.swift

```diff
@@ -0,0 +1,214 @@
+                
+                operation(target, property.name)
+            case let dynamicProperty as DynamicProperty:
+                assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "DynamicProperty \(property.name) on element \(info.name) must be a struct")
+
+                dynamicProperty.execute(operation)
+            case let dynamics as Traversible:
+                assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "Dynamics \(property.name) on element \(info.name) must be a struct")
+            
+                dynamics.execute(operation)
+            default:
+                break
+            }
+        }
+    } catch {
+        fatalError("Applying operation on all properties of \((try? typeInfo(of: Target.self))?.name ?? "Unknown Type") on element \(element) failed.")
+    }
+}
+
+func execute<Element, Target>(_ operation: (Target) -> Void, on element: Element) {
+    execute({(target: Target, _: String) in
+        operation(target)
+    }, on: element)
+}
+
+// MARK: Apply
+func apply<Element, Target>(_ mutation: (inout Target, _ name: String) -> Void, to element: inout Element) {
+    do {
+        let info = try typeInfo(of: Element.self)
+
+        for property in info.properties {
+            let child = try property.get(from: element)
+
+            switch child {
+            case var target as Target:
+                assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "\(Target.self) \(property.name) on element \(info.name) must be a struct")
+                
+                mutation(&target, property.name)
+                try property.set(value: target, on: &element)
+            case var dynamicProperty as DynamicProperty:
+                assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "DynamicProperty \(property.name) on element \(info.name) must be a struct")
+                dynamicProperty.apply(mutation)
+                try property.set(value: dynamicProperty, on: &element)
+            case var dynamics as Traversible:
+                assert(((try? typeInfo(of: property.type).kind) ?? .none) == .struct, "Dynamics \(property.name) on element \(info.name) must be a struct")
+            
+                dynamics.apply(mutation)
+                try property.set(value: dynamics, on: &element)
+            default:
+                break
+            }
+        }
+    } catch {
+        fatalError("Applying operation on all properties of \((try? typeInfo(of: Target.self))?.name ?? "Unknown Type") on element \(element) failed.")
+    }
+}
+
+func apply<Element, Target>(_ mutation: (inout Target) -> Void, to element: inout Element) {
+    apply({(target: inout Target, _: String) in
+        mutation(&target)
+    }, to: &element)
+}
+
+// MARK: DynamicProperty Implementation
+private extension DynamicProperty {
```

Comment:
There is a high code duplication between this implementation as a private extension of `DynamicProperty` and the global function that I would suggest to reduce. I would propose to move all the functionality in a single `RequestInjectable` or `Injectable` protocol extension as detailed above and it might even be reasonable to move it to an `inject` function that can be used by the semantic model.
The same code duplication could also be seen in the `apply` function.

Reply:
I agree on the part about the code-duplication. I tried to reduce this a couple of times, but failed because of corrupted memory access. I would be happy to address this problem in a pair-programming session with you or @nerdsupremacist.